sn_client/src/files/upload.rs

```diff
@@ -186,210 +218,175 @@ impl FilesUpload {
         result
     }
 
-    async fn upload(&mut self, chunks: Vec<(XorName, PathBuf)>) -> Result<()> {
+    /// There are three main task groups to upload chunks:
+    ///   1, Fetch the store_cost of a chunk
+    ///   2, Pay for the chunk based on the fetched store_cost
+    ///   3, Upload the chunk
+    /// Task groups can be parallely to each other, however requires input from previous group.
+    /// Within each group, for group 1 and 3, the mini tasks inside can be parallel to each other.
+    /// However for group 2, mini tasks inside has to be undertaken sequencely.
+    /// Hence, the flow is more suitable to be handled in a `parrallelized pipline` way,
+    /// instead of the current `batched pipline` way.
```

Comment:
This is fair commentary on the PR, but doesnt make sense as doc comments, 

Lets just clarify what's going on in the fn.

I think this is fine: 
```
/// There are three main task groups to upload chunks:
    ///   1, Fetch the store_cost of a chunk
    ///   2, Pay for the chunk based on the fetched store_cost
    ///   3, Upload the chunk
    /// Task groups can be run in parallel to each other, however sometimes requires input from previous group.
    /// Within each group, for group 1 and 3, the mini tasks inside can be parallel to each other.
    /// However for group 2, mini tasks inside has to be undertaken sequentially.```

(note there's some typo corrections there)